test/inject.test.js

```diff
@@ -409,3 +409,24 @@ test('should throw error if callback specified and if ready errors', t => {
     t.strictEqual(err, error)
   })
 })
+
+test('should support builder-style injection', t => {
+  t.plan(3)
+  const fastify = Fastify()
+  const payload = { hello: 'world' }
+
+  fastify.get('/', (req, reply) => {
+    reply.send(payload)
+  })
+
+  fastify.ready()
+    .then(() => {
+      fastify
+        .inject().get('/').end()
+        .then((res) => {
+          t.deepEqual(payload, JSON.parse(res.payload))
+          t.strictEqual(res.statusCode, 200)
+          t.strictEqual(res.headers['content-length'], '17')
+        })
+    })
+})
```

Comment:
Could you add another test without the `.ready().then()`?

You can also switch this to async/await, as this is a v3 specific bug.

Reply:
Done.